### ISSUE TITLE
Issue with SYS_FUNC_HALT_POW_OFF on RPi 4 + Raspberry Pi OS 64

### ISSUE THREAD

**Issue by smallhacks** [PiSupply/PiJuice#772]
Hello,

I managed to install the PiJuice software on Raspberry Pi OS 64 bit on a Raspberry Pi 4. Everything works well, except that the SetPowerOff method triggered by SYS_FUNC_HALT_POW_OFF seems to be not working correctly.

I configured a button to trigger SYS_FUNC_HALT_POW_OFF. The normal behaviour is to shutdown the RPi, wait for 60 sec and then power off the RPi. The power led of the RPi is ON and then turns off when the RPi is powered off. I tried on vanilla Raspberry Pi OS 32 bit and vanilla Ubuntu 64 bit and it works as expected.

But on Raspberry Pi OS 64 bit, the shutdown process seems to start correctly (the activity led is blinking), but after only 10 seconds, the activity led and the power led suddenly turns off.

I tried on two different RPi 4 and two different PiJuice. I also tried to downgrade the firmware of the PiJuice from V1.5 to V1.4 and it doesn’t’ work either.

Any help would be appreciated.

**Comment by tvoverbeek** [comment]
I have no Raspberry Pi Os 64-bit running here.
The SYS_FUNC_HALT_POW_OFF is executed by the pijuice service (pijuice_sys.py) which is a python script.
See https://github.com/PiSupply/PiJuice/blob/master/Software/Source/src/pijuice_sys.py lines 47-70.
Check your copy of pijuice_sys.py in /usr/bin if line 68 specifies 60 seconds as argument.

**Comment by smallhacks** [comment]
I checked my copy of pijuice_sys.py and SetPowerOff is set to 60 seconds (I did not change the original script file).

Here are the difference in the shutdown sequence between RPi OS 32 and 64 I noticed:
**RPi OS 32 and Ubuntu 64**
1. The activity led blinks according to the CPU activity or card writing
2. The power led turns of for a few [milliseconds]
3. The power led turns on
3. The activity led blinks 10 times

**RPi OS 64**
1. The activity led blinks according to the CPU activity or card writing
2. The power led turns of but never turns on again

I made two videos: [Raspberry Pi OS 32](https://youtu.be/J8YfqbusmWM) and [Raspberry Pi OS 64](https://youtu.be/x5-vXy2JJlU).

**Comment by tvoverbeek** [comment]
Are these on a Pi4? The Pi4 has a configurable boot loader.
Are you up-to-date with the boot loader?
See https://www.raspberrypi.org/documentation/computers/raspberry-pi.html#:~:text=available%20on%20Github.-,Raspberry,-Pi%204%20Boot
There is an option which turns the power off on system halt, which maybe is what you are seeing.

**Comment by smallhacks** [comment]
Yes, this is on a Pi4.
I updated the boot loader and checked the configuration. The POWER_OFF_ON_HALT is OFF as you can see:


    BOOT_UART=0
    WAKE_ON_GPIO=1
    POWER_OFF_ON_HALT=0

**Comment by tvoverbeek** [comment]
If you do a normal shutdown from the terminal or GUI (when running on power to the Pi4, no PiJuice connected) do you get the same behavior or do you see the same as on the 32-bit OS?

**Comment by smallhacks** [comment]
If I do a normal shutdown from the terminal, I get the expected behavior (same as on the 32-bit OS, power led turns ON again). Even if the PiJuice is plugged.

**Comment by tvoverbeek** [comment]
When the pijuice service shuts down the RPi it uses `sudo halt`.
Cna you run `sudo halt` from a terminal and see hat behavior you get?
The 64-bit os might do different things for 'shutdown', 'poweroff' and 'halt'.

**Comment by tvoverbeek** [comment]
OK, I installed a copy of Raspberry Pi OS 64-bit on one of my Pi4-s today.
Ran apt update and apt upgrade and am now on kernel:
`Linux rpi64 5.10.52-v8+ #1441 SMP PREEMPT Tue Aug 3 18:14:03 BST 2021 aarch64`
Shutting down with `sudo halt` has the behavior you observed (No 10 flashes, red LED stays off).
Using `sudo poweroff` or `sudo shutdown` gives the 32-bit behavior (10 flashes after the initial LED off and red LED stays on as long as external power is connected).
So not caused by the PiJuice but an 64-bit OS problem/property
